cumulus/pallets/parachain-system/src/lib.rs

```diff
@@ -1479,9 +1538,20 @@ impl<T: Config> Pallet<T> {
 		//
 		// However, changing this setting is expected to be rare.
 		if let Some(cfg) = Self::host_configuration() {
-			if message.len() > cfg.max_upward_message_size as usize {
+			if message_len > cfg.max_upward_message_size as usize {
 				return Err(MessageSendError::TooBig)
 			}
+			let threshold = cfg.max_upward_queue_size.saturating_div(ump_constants::THRESHOLD_FACTOR);
+			// We check the threshold against total size and not number of messages since messages
+			// could be big or small
+			let pending_messages = PendingUpwardMessages::<T>::get();
+			let total_size: usize = pending_messages.iter().map(UpwardMessage::len).sum();
+			if total_size > threshold as usize {
+				let message_size_factor =
+					FixedU128::from_u32(message_len.saturating_div(1024) as u32)
+						.saturating_mul(ump_constants::MESSAGE_SIZE_FEE_BASE);
+				Self::increment_ump_fee_factor(message_size_factor);
```

Comment:
I think this code could generally use some more comments, either in-line or in a new doc comment for the method (currently missing)?